services/pom.xml

```diff
@@ -107,6 +107,42 @@
                         </execution>
                     </executions>
                 </plugin>
+                <plugin>
+                    <groupId>com.github.spotbugs</groupId>
+                    <artifactId>spotbugs-maven-plugin</artifactId>
+                    <version>3.1.11</version>
+                    <executions>
+                        <execution>
+                            <goals>
+                                <goal>check</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <configuration>
+                        <effort>Max</effort>
+                        <threshold>High</threshold>
+                        <xmlOutput>true</xmlOutput>
+                        <excludeFilterFile>${project.basedir}/../spotbugs-exclude.xml</excludeFilterFile>
```

Comment:
Doesn't this leak parent files into the services folder? If we move the services dir to a new repo, this would stop working, for example.

Reply:
Oh, nm, I see. This is relative to the sub-dir of services. That's been flaky for us in the past. Might be nicer if the file could be pulled in from a lib or something. Or there's a plugin for determining the dir of a parent pom.

Reply:
No this allows the child modules under `services/` dir to read `services/spotbugs-exclude.xml`

I had considered having a single shared excludes file, but opted not to for exactly the reason you mentioned. Figured it was better to keep root modules and service modules completely independent for this

Reply:
Sorry didn't see your follow-up for some reason. Yeah, this won't work for any future modules that get nested as grandchildren, etc, etc of services. Mulling over options

Reply:
According to their [javadoc](https://github.com/spotbugs/spotbugs-maven-plugin/blob/spotbugs/src/main/groovy/org/codehaus/mojo/spotbugs/SpotBugsMojo.groovy#L278) , the filter files can be resolved from classpath, URI, and file path.

So, I guess we could give each jar it's own distinct _excludes_ file, or have one jar/one file that we add to each module's classpath when the spotbugs profile is activated. Not sure I love either of those options.

Wouldn't it be less intrusive just to use the same trick that we use for _${datawave.root}_, e.g., _${services.root}_ ?